### PR TITLE
Fix size of data copied to constant memory and add missing include

### PR DESCRIPTION
**Pull request by Rombur**
This PR fixes most of the failing CUDA tests.

### CODE REVIEW

File: include/deal.II/matrix_free/cuda_matrix_free.templates.h [hunk 1]
```diff
@@ -274,7 +274,9 @@ namespace CUDAWrappers
       cudaError_t error_code = cudaMemcpyToSymbol(
         constraint_weights,
         shape_info.data.front().subface_interpolation_matrix.data(),
-        sizeof(double) * fe.n_dofs_per_face(0) * fe.n_dofs_per_face(0));
+        sizeof(double) *
+          shape_info.data.front().subface_interpolation_matrix.size());
```
Comment (peterrum) [inline include/deal.II/matrix_free/cuda_matrix_free.templates.h:278]: Thanks! But shouldn't that be the same?

Comment (Rombur) [reply include/deal.II/matrix_free/cuda_matrix_free.templates.h:278]: That's what I thought too but it's not. It is the same in 2D but not in 3D.

Comment (peterrum) [reply include/deal.II/matrix_free/cuda_matrix_free.templates.h:278]: Do you happen to have an example of values? I'll take a look at that later. Independent of this, this change is fine to merge!

**Review by peterrum — APPROVED** [review]

File: include/deal.II/matrix_free/cuda_matrix_free.templates.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/matrix_free/cuda_matrix_free.templates.h:278]: Right, these are 1D interpolation matrices. Makes all sense! Thanks for the fix!

**Review by masterleinad — APPROVED** [review]

File: include/deal.II/matrix_free/cuda_matrix_free.templates.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/matrix_free/cuda_matrix_free.templates.h:278]: It's certainly more consistent this way to get the pointer and the number of elements from the same place.

**Review by bangerth — APPROVED** [review]

**Comment by marcfehling** [comment]
Do we need to run the `jenkins CI`? AFAIK we do not check for CUDA support there, but in the end it doesn't hurt to run it once more...

/rebuild

**Review by kronbichler — APPROVED** [review]

**Pull request merged**